### DISCUSSION TITLE
Woks with Deye SUN-M80G3-EU-Q0

### DISCUSSION THREAD

**Discussion by DonKracho** [StephanJoubert/home_assistant_solarman#350] [General]
@StephanJoubert Thank you for this great Solarman integration👍 I just want to mention that it worked for me with HomeAssistant 2023.7.2 and a Deye SUN-M80G3-EU-Q0 microinverter (successor of the SUN-800) The Serial#No is starting with 39xx and the FW version is MW3_16U_5406_1.53, I did not upgrade it jet. As recommended I used the deye_2mppt.yaml definition. I did manually install the solarman integration, because I do not want to have HACS installed on a productive HomeAssistant installation.

**Comment by TheFreaker86** [comment]
Hallo,
I'm new to HA and I try to get the same microinverter integrated. I installed the integration manually like you. The inverter SN starts with 23 and the logger SN starts with 39 like yours. and I configured it according to the instructions, using the SN 39xxxxxxxx. I installed the integration manually like you. 

I tried the following thing:

- I first had the firmware version 1.57 installed, went then to version MW3_16U_5406_2.06 and after reading your post I also tried MW3_16U_5406_1.53. Now i'm back at 2.06
- I tried to comment out the validation part like in #325 (the only difference being I edited the deye_2mppt.yaml)
- I confirmed by portscan that the port 8899 is open on the inverter
- I used the autodiscover feature by configuring the IP adress 0.0.0.0
- I tried setting several inverter types (like deye_4mppt.yaml or deye_string.yaml).
 
But nothing works. The only reading I get displayed is the inverter temperature. Everything else is shown as unknown or not available. The deye_string.yaml gives some more readings but it's not everthing I need.

I created a debug log file (see attachment), but can't make anything of it:
[home-assistant_solarman_2023-08-30T11-57-03.867Z.log](https://github.com/StephanJoubert/home_assistant_solarman/files/12477200/home-assistant_solarman_2023-08-30T11-57-03.867Z.log)

Home Assistant Version: 
OS: 10.5
Core: 2023.8.4

Can you please point me in the right direction?

**Reply by TheFreaker86** [reply]
Never mind. I just discovered that evcc configure offers a template for those Deye microinverters. Maybe that template was included in the evcc update that went public today. Works fine.

**Reply by Heidieker** [reply]
Hi @TheFreaker86 i am at the same point, that the SUN-M80G3-EU-Q0 ist noct connecting in hass. But i dont find the evcc template. Can you help me?

**Reply by TheFreaker86** [reply]
Ahoi! 
this is what I have in my config for EVCC:

```
meters:
- type: template
  template: solarman
  usage: pv
  host: HOSTNAME OR IP ADRESS
  user: USERNAME
  password: PASSWORD
  name: NAME
```
  
  change the values as needed.

**Reply by luck0r** [reply]
i used deye_2mppt.yaml for my SUN-M80G3-EU-Q0. Worked out of the box with the setup dialog

**Comment by workale** [comment]
I would also be interested in integrating this inverter into HA, it would be very interesting to be able to send remote controls to the micro inverter to lower the power based on consumption read by a Power Meter, does anyone know if this can be done or has it been done?

**Comment by chrossi2000** [comment]
Hi everybody,
I'm still a beginner with HA and HACS, but I want to share how the solarman integration finally worked for me.
I kept getting no readings althoug the invester was connected via wifi and the solarman app showed results.
So I checked on the Deye Webinterface and there whas no data in the inverter information.
I then ran the Wifi Wizard again, although the connection was running (read this thrick somewhere else).
Afterwards the inverter information on the Deye Webinterface was updated and shortly after the data also showed up in HA.

Hope this helps some of you.

My hardware: Deye SUN-M80G3-EU-Q0
Firmware: MW3_SSL_5408_1.0B
Logger SN: 39xxxxxx
Port: 8899
Lookup File: deye_string.yaml

**Reply by pagea-dev** [reply]
you need to use deye_2mppt.yaml when you have the Deye SUN-M80G3-EU-Q0 - if you dont see the values within a few minutes. just start HASS again.

**Reply by chrossi2000** [reply]
Thanks, works better with deye_2mppt.yaml

**Comment by msman0** [comment]
Hi, I have also a Deye SUN-M80G3-EU-Q0 and set up in HA with deye_2mppt.yaml. I don't get any values. My serial number (from web interface) starts with 9.. 
What is wrong with the configuration?
Thanks.

**Reply by msman0** [reply]
It works now. Not the serial number from web interface is correct I must choose the number from the label.

**Comment by holger-wi** [comment]
It works really well but I'm missing the actual power of each channel. Calculating ist not really accurate.
Is it possible to read it from the Microinverter and implement it into the 2mppt setup?
Thanks a lot for the work.
Holger-wi

**Comment by schu4you** [comment]
It works only with the deye default admin credentials, when you change the password and username it didn`t work

**Reply by holger-wi** [reply]
Hello shu4you,
I think there is a missunderstanding. 
38 Entities are read. See screenshot of some entities. But it would be perfekt to see the actual power ov PV1 and PV2. The microinverter provides this values. In iobroker it can be read ;-)
![Screenshot_20251004_095125_Chrome](https://github.com/user-attachments/assets/c6326a59-39af-48f7-a313-734d390a4e5c)
